### DISCUSSION TITLE
Replace MenuComplete with ListView

### DISCUSSION THREAD

**Discussion by totkeks** [lzybkr/PSReadLine#3475] [Ideas]
I just started using the new prediction option and I like the ListView much more than the old MenuComplete. Is it planned to replace MenuComplete or add the ListView option to the regular completion?

**Comment by StevenBucher98** [comment]
@totkeks very interesting idea! So would you like to see something like as you are typing your cmdlets the list view shows the possible options as you type them?

**Reply by totkeks** [reply]
Yes, like the predictions here (https://devblogs.microsoft.com/powershell/announcing-psreadline-2-1-with-predictive-intellisense/) and the predicted completions here (https://github.com/PowerShell/CompletionPredictor), except it is the real completions, that I would normally get using tab and this setting `Set-PSReadLineKeyHandler -Chord Tab -Function MenuComplete`.

Regular MenuComplete
![image](https://user-images.githubusercontent.com/2941389/200432968-f3175ab0-5f3b-4af0-9503-dbe218affae9.png)

Should be displayed like this
![image](https://github.com/PowerShell/CompletionPredictor/raw/main/tools/images/CompletionPredictor.gif)

**Comment by ThomasNieto** [comment]
I do not want to see MenuComplete replaced with ListView. MenuComplete has tooltips and can show more items than ListView. Each view has its own purpose and people who prefer one over the other. There are times where I use ListView and times when I use MenuComplete.

![image](https://user-images.githubusercontent.com/38873752/200481186-e619991a-7a89-4bb9-9370-f154fee58ca4.png)

**Comment by totkeks** [comment]
Thanks for the feedback, Thomas! You are right, those tooltips are useful and should be added to the ListView as well.

Maybe I should rephrase my suggestion then and rather say, let us choose if we want MenuComplete or ListView for the regular completions.

**Reply by ThomasNieto** [reply]
Please correct me if I'm wrong but your request is to be able to change the way completions are presented either in a MenuComplete style with multiple columns and tooltip or ListView style with all the results in a single row with tooltip at bottom. Is that correct?

**Reply by hrxn** [reply]
@ThomasNieto I assume the suggestion is to display the completions like this: https://github.com/PowerShell/CompletionPredictor

Albeit the remaining question remains the placement of the tooltip from classic MenuComplete (which is indeed very useful, I agree with your point here)? Below the result list, or maybe a new column?

@totkeks Forgive me if interpret this the wrong way, but what you are basically asking is, that https://github.com/PowerShell/CompletionPredictor should be integrated as-is (more or less) into default PSReadLine, right?

If so, I think this is a proposal worth of consideration..

**Reply by totkeks** [reply]
@ThomasNieto sorry for the late reply. Yes, that would be my request. Basically to have completions behave identical if configured that way.

@Hrxn yes, that's it basically. And right, the tooltip should be added to list view as it is quite useful. 

So the result should be that I can configure the completion style in psreadline to either menu view or list view, and all completion "tools" work the same way.

As an example, when I set psreadline to menu complete then I get the menu view both for regular completions, like currently, as well as the new predictors. 

When I set psreadline to list view, then I get the list view for the regular completions and for the predictors, like currently. 

Or, if you want to complicate things a bit, allow different view settings for different completion "types". Though that is probably not useful and just creates unnecessary work.
